README.md

```diff
@@ -95,76 +206,71 @@ def rots(d, s, tree):
 
-(...)
-```
-
-This
-[file](https://gist.github.com/VictorTaelin/face210ca4bc30d96b2d5980278d3921)
-implements a [bitonic sorter](https://en.wikipedia.org/wiki/Bitonic_sorter) with
-*immutable tree rotations*. It is not the kind of algorithm you'd expect to
-run fast on GPUs. Yet, since it uses a divide-and-conquer approach, which is
-*inherently parallel*, Bend will run it multi-threaded. Some benchmarks:
-
-- CPU, Apple M3 Max, 1 thread: **12.15 seconds**
-
-- CPU, Apple M3 Max, 16 threads: **0.96 seconds**
-
-- GPU, NVIDIA RTX 4090, 16k threads: **0.21 seconds**
-
-That's a **57x speedup** by doing nothing. No thread spawning, no explicit
-management of locks, mutexes. We just asked Bend to run our program on RTX, and
-it did. Simple as that.
-
-Bend isn't limited to a specific paradigm, like tensors or matrices. Any
-concurrent system, from shaders to Erlang-like actor models can be emulated on
-Bend. For example, to render images in real time, we could simply allocate an
-immutable tree on each frame:
-
-```python
-# given a shader, returns a square image
-def render(depth, shader):
-  bend d = 0, i = 0:
-    when d < depth:
-      color = (fork(d+1, i*2+0), fork(d+1, i*2+1))
-    else:
-      width = depth / 2
-      color = shader(i % width, i / width)
-  return color
-
-# given a position, returns a color
-# for this demo, it just busy loops
-def demo_shader(x, y):
-  bend i = 0:
-    when i < 5000:
-      color = fork(i + 1)
-    else:
-      color = 0x000001
-  return color
-
-# renders a 256x256 image using demo_shader
+# Swaps distant values (Red Box)
+def warp(d, s, a, b):
+  switch d:
+    case 0:
+      return swap(s ^ (a > b), a, b)
+    case _:
+      (a.a, a.b) = a
+      (b.a, b.b) = b
+      (A.a, A.b) = warp(d-1, s, a.a, b.a)
+      (B.a, B.b) = warp(d-1, s, a.b, b.b)
+      return ((A.a,B.a),(A.b,B.b))
+
+# Propagates downwards
+def down(d,s,t):
+  switch d:
+    case 0:
+      return t
+    case _:
+      (t.a, t.b) = t
+      return (rots(d-1, s, t.a), rots(d-1, s, t.b))
+
+# Swaps a single pair
+def swap(s, a, b):
+  switch s:
+    case 0:
+      return (a,b)
+    case _:
+      return (b,a)
+
+# Testing
+# -------
+
+# Generates a big tree
+def gen(d, x):
+  switch d:
+    case 0:
+      return x
+    case _:
+      return (gen(d-1, x * 2 + 1), gen(d-1, x * 2))
+
+# Sums a big tree
+def sum(d, t):
+  switch d:
+    case 0:
+      return t
+    case _:
+      (t.a, t.b) = t
+      return sum(d-1, t.a) + sum(d-1, t.b)
+
+# Sorts a big tree
 def main:
-  return render(16, demo_shader)
+  return sum(20, sort(20, 0, gen(20, 0)))
 ```
 
-And it would actually work. Even involved algorithms parallelize well on Bend.
-Long-distance communication is performed by *global beta-reduction* (as per the
-[Interaction Calculus](https://github.com/VictorTaelin/Interaction-Calculus)),
-and synchronized correctly and efficiently by
-[HVM2](https://github.com/HigherOrderCO/HVM)'s *atomic linker*.
-
+#### Benchmark
```

Comment:
This section is only shown when users click the summary. You wanted only the code to be collapsed right?

Reply:
Fixed